opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java

```diff
@@ -150,14 +159,46 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 IndividualProfileRemoveActivity.startIndividualProfileActivity(ChildProfileActivity.this, presenter().getChildClient(),
                         ((ChildProfilePresenter) presenter()).getFamilyID()
                         , ((ChildProfilePresenter) presenter()).getFamilyHeadID(), ((ChildProfilePresenter) presenter()).getPrimaryCareGiverID(), ChildRegisterActivity.class.getCanonicalName());
-
                 return true;
+            case R.id.action_thinkmd_health_assessment:
+                try {
+                    ThinkMDLibrary.getInstance().processHealthAssessment(prepareFHIRModel());
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
             default:
                 break;
         }
         return super.onOptionsItemSelected(item);
     }
 
+    private FHIRBundleModel prepareFHIRModel() {
+        FHIRBundleModel model = new FHIRBundleModel();
+        model.setRandomlyGeneratedId(getRandomGeneratedId());
+        model.setEncounterId(getRandomGeneratedId());
+        model.setRootPackageName(getApplicationContext().getPackageName());
+        model.setAppVersion(BuildConfig.VERSION_NAME);
+        model.setDisplayLanguage(getDisplayLanguage(this));
+        model.setAppName(getResources().getString(R.string.app_name));
+        model.setAppLanguage(getAppLanguage(this));
+        Triple<String, String, String> userProfile = fetchUserProfileFromDB(childBaseEntityId);
+        model.setGender(userProfile.getRight());
+        model.setDob(userProfile.getMiddle());
+        model.setAgeInDays(userProfile.getLeft());
+        Pair<String, String> muacPair = fetchMUACFromDB(childBaseEntityId);
+        model.setMUACValueCode(muacPair.first);
+        model.setMUACValueDisplay(muacPair.second);
+        //Todo: these values needs to be query and set into model
+        model.setPractitionerId(null);
+        model.setPatientId(null);
+        model.setUserName(null);
+        model.setLocationId(null);
+        model.setUniqueIdGeneratedForThinkMD(null);
+        model.setEndPointPackageName(null);
+
+        return model;
+    }
```

Comment:
Define this in a class as a method that be overriden by a CHW flavour. You also need to pass the `childBaseEntityId` to the method

Reply:
The base Implementation can be moved also to CHW core library

Reply:
Done